fog/view/enclave/impl/src/types.rs

```diff
@@ -884,4 +916,366 @@ mod get_block_data_tests {
             last_fully_processed_timestamp
         );
     }
+
+    #[test]
+    fn first_response_incomplete_missing_blocks_cover_it() {
+        const STORE_COUNT: usize = 4;
+        let mut decrypted_query_responses = Vec::with_capacity(STORE_COUNT);
+
+        // Make the first response only process block at index 0 (block count is 1).
+        // Make the missed_block_range includes blocks at index 1 and 2. Since this
+        // ingest is not responsible for them, it should not factor into the
+        // highest processed block count calculation.
+        let incomplete_block_count = 1;
+        let incomplete_timestamp = 1;
+        let missed_block_range = BlockRange::new(1, 3);
```

Comment:
I added the test you describe for each one, but we actually need this test as well. Without it, we wouldn't be certain that we are NOT counting missed blocks in the final calculation.

Reply:
FWIW, it's important to write test cases not solely based on the implementation's internals (e.g. if statements) because those change. There are also test cases that need to be added based on known behavior.

Reply:
I agree about testing behavior not implementation.
And I agree that there should be a test case for NOT counting missed blocks in the final calculation, I'm just not sure we need to test that for first block, second block and penultimate block.  I think testing it for any block that isn't the last would suffice.